### PR TITLE
🐛 ⚡ (Bugfix) Bots no longer created when clicking create button #74

### LINKED ISSUE #74: 🐛⚡ Bots no longer created when clicking create button
**Describe the bug**
On the user dashboard (bots list page), when clicking create bot the bots are no longer created.

**To Reproduce**
1. Go to the user dashboard
image

2. Click on 'New bot', you'll see the below:

image

**PROBLEM 1 - DEFAULT NAME OF BOT SHOULD BE FILLED**
A default name for the bot should be added. Create a "generate name" function in the newstory.service.ts

**PROBLEM 2 - Scroll icons**

The modal shows a scroll-left and scroll-right icon. Those shouldn't be there.

image

**PROBLEM 3 - Translations**
Please add translations for the modal in the app/assets/en.json and fr.json files

3. Create the bot by clicking the create button

**PROBLEM 4 - Bot not added**
The bot is not created. Use the newstory.service to make this happen.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by AnnaL001**
# Description

I have edited the modal "name" form input to have a filled-in default value by using the ReactiveFormsModule. Aside from that, I have added translations for the modal as specified. Finally, I have updated the create bot functionality to allow for the creation of bots. 

Fixes #74 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# Screenshot (optional)
Before
Screenshot from 2022-10-04 15-32-20

After
Screenshot from 2022-10-05 02-53-21
Screenshot from 2022-10-05 02-54-29
Screenshot from 2022-10-05 02-52-44

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/home/src/lib/pages/home/home.page.ts [hunk 1]
```diff
@@ -48,18 +46,16 @@ export class HomePageComponent implements OnDestroy
...
     this._sb.unsubscribe();
   }
 
-  // openDialog(){
-  //   this.dialog.open(CreateBotModalComponent,{
-  //     height: '480px',
-  //     width: '600px',
+  openDialog(){
+    this.dialog.open(CreateBotModalComponent,{
+      height: '480px',
```
Comment (ReaganCn) [inline libs/features/convs-mgr/home/src/lib/pages/home/home.page.ts:56]: You can move this styling to the CreateModalComponent scss.

File: libs/features/convs-mgr/home/src/lib/modals/create-bot-modal/create-bot-modal.component.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+import { NewStoryService } from '../../services/new-story.service';
 
 @Component({
   selector: 'convl-italanta-apps-create-bot-modal',
   templateUrl: './create-bot-modal.component.html',
   styleUrls: ['./create-bot-modal.component.scss'],
 })
 export class CreateBotModalComponent implements OnInit {
-  constructor() {}
+  botForm = this._formBuilder.group({
```
Comment (ReaganCn) [inline libs/features/convs-mgr/home/src/lib/modals/create-bot-modal/create-bot-modal.component.ts:12]: Wrap the botForm formBuilder in a method and place it below the constructor. This ensures clean code.

File: libs/features/convs-mgr/home/src/lib/services/new-story.service.ts [hunk 3]
```diff
@@ -3,35 +3,46 @@ import { take, map, switchMap, tap } from "rxjs/operators";
...
-              private _router: Router)
+              private _router: Router,
+              private dialog: MatDialog)
   {}
 
-  add(name?: string) {
+  add(name?: string, description?: string) {
     // Generate default name if name not passed.
     if(!name)
       name = uniqueNamesGenerator({ dictionaries: [adjectives, colors, animals] });
```
Comment (ReaganCn) [inline libs/features/convs-mgr/home/src/lib/services/new-story.service.ts:27]: Call the generate name method here instead of `uniqueNamesGenerator`

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Congrats on your first PR 🎉, and a good first attempt. I have requested some straight forward changes on your code. Please see my comments below.

File: libs/features/convs-mgr/home/src/lib/modals/create-bot-modal/create-bot-modal.component.ts (hunk 2 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/home/src/lib/modals/create-bot-modal/create-bot-modal.component.ts]: Change has been made

File: libs/features/convs-mgr/home/src/lib/pages/home/home.page.ts (hunk 1 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/home/src/lib/pages/home/home.page.ts]: Change has been made

**Review by ReaganCn — APPROVED** [review]
Good job 💪!!!

**Pull request merged**